packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -176,16 +182,20 @@ export class VirtualNode extends VirtualNodeBase {
     props.listeners?.forEach(listener => this.addListener(listener));
     const accessLogging = props.accessLog?.bind(this);
     const serviceDiscovery = props.serviceDiscovery?.bind(this);
+    const tlsClientPolicy = props.backendDefaults?.tlsClientPolicy;
 
     const node = new CfnVirtualNode(this, 'Resource', {
       virtualNodeName: this.physicalName,
       meshName: this.mesh.meshName,
       spec: {
         backends: cdk.Lazy.any({ produce: () => this.backends }, { omitEmptyArray: true }),
         listeners: cdk.Lazy.any({ produce: () => this.listeners.map(listener => listener.listener) }, { omitEmptyArray: true }),
-        backendDefaults: props.backendDefaults !== undefined
+        backendDefaults: props.backendDefaults
           ? {
-            clientPolicy: props.backendDefaults?.clientPolicy?.bind(this).clientPolicy,
+            clientPolicy: {
+              tls: tlsClientPolicy ? renderTlsPolicy(this, tlsClientPolicy,
+                (config) => config.virtualNodeClientTlsValidationTrust) : undefined,
+            },
           }
           : undefined,
         serviceDiscovery: {
```

Comment:
Same comments here (leave the previous condition unchanged, making `renderTlsPolicy()` handle the `undefined` case simplifies this code).

Reply:
Responded [here](https://github.com/aws/aws-cdk/pull/14782#discussion_r637160804).

Reply:
Addressing this as well.